File: LeetCode/java/1897_Redistribute_Characters_to_Make_All_Strings_Equal.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public boolean makeEqual(String[] words) {
        int wordCount = words.length;
        Map<Character, Integer> map = new HashMap<>();

        for (int i = 0; i < words.length; i++) {
            for (int j = 0; j < words[i].length(); j++) {
                map.put(words[i].charAt(j), map.getOrDefault(words[i].charAt(j), 0) + 1);
            }
        }

        for (Character key : map.keySet()) {
            if (map.get(key) % wordCount != 0) {
                return false;
            }
        }

        return true;
    }
}

File: LeetCode/cpp/1897_Redistribute_Characters_to_Make_All_Strings_Equal.cpp
class Solution {
public:
	bool makeEqual(vector<string>& words) {
		int word_count = words.size();
		unordered_map<char, int> um;

		for (int i = 0; i < words.size(); i++) {
			for (int j = 0; j < words[i].size(); j++) {
				um[words[i][j]]++;
			}
		}

		for (auto m : um) {
			if (m.second % word_count != 0) {
				return false;
			}
		}

		return true;
	}
};